Sources/PubNub/Helpers/Crypto/CryptorModule.swift

```diff
@@ -0,0 +1,320 @@
+        ).asData() + $0.data
+      }
+    }.mapError {
+      PubNubError(.encryptionError, underlying: $0)
+    }
+  }
+  
+  /// Decrypts the given `Data` object
+  ///
+  /// - Parameters:
+  ///   - data: Data to encrypt
+  /// - Returns: A success, storing decrypted `Data` if operation succeeds. Otherwise, a failure storing `PubNubError` is returned
+  public func decrypt(data: Data) -> Result<Data, PubNubError> {
+    guard !data.isEmpty else {
+      return .failure(PubNubError(
+        .decryptionError,
+        additional: ["Cannot decrypt empty Data"])
+      )
+    }
+    do {
+      let header = try CryptorHeader.from(data: data)
+      
+      guard let cryptor = cryptor(matching: header) else {
+        return .failure(PubNubError(
+          .unknownCryptorError,
+          additional: [
+            "Could not find matching Cryptor for \(header.cryptorId()) while decrypting Data. " +
+            "Ensure the corresponding instance is registered in \(String(describing: Self.self))"
+          ]
+        ))
+      }
+      return cryptor.decrypt(
+        data: EncryptedData(
+          metadata: header.metadataIfAny(),
+          data: data.subdata(in: header.length()..<data.count)
+        )
+      )
+      .flatMap {
+        if $0.isEmpty {
+          return .failure(PubNubError(
+            .decryptionError,
+            additional: ["Decrypting resulted with empty Data"])
+          )
+        }
+        return .success($0)
+      }
+      .mapError {
+        PubNubError(.decryptionError, underlying: $0)
+      }
+    } catch let error as PubNubError {
+      return .failure(error)
+    } catch {
+      return .failure(PubNubError(
+        .decryptionError,
+        underlying: error,
+        additional: ["Cannot decrypt InputStream"])
+      )
+    }
+  }
+  
+  /// Encrypts the given `InputStream` object
+  ///
+  /// - Parameters:
+  ///   - stream: Stream to encrypt
+  ///   - contentLength: Content length of encoded stream
+  /// - Returns: A success, storing an ``EncryptedStreamResult`` value if operation succeeds. Otherwise, a failure storing `PubNubError` is returned
+  public func encrypt(stream: InputStream, contentLength: Int) -> Result<EncryptedStreamResult, PubNubError> {
+    guard contentLength > 0 else {
+      return .failure(PubNubError(
+        .encryptionError,
+        additional: ["Cannot encrypt empty InputStream"])
+      )
+    }
+    return defaultCryptor.encrypt(
+      stream: stream,
+      contentLength: contentLength
+    ).map {
+      let header = defaultCryptor.id != LegacyCryptor.ID ? CryptorHeader.v1(
+        cryptorId: defaultCryptor.id,
+        data: $0.metadata
+      ) : .none
+      
+      let multipartInputStream = MultipartInputStream(
+        inputStreams: [InputStream(data: header.asData()), $0.stream]
+      )
+      return EncryptedStreamResult(
+        stream: multipartInputStream,
+        contentLength: $0.contentLength + header.length()
+      )
```

Comment:
I would return an instance of our `MultipartInputStream` here, and it will be transparent for use (`InputStream`).
In send file request configuration, we need the length of the stream (user don't need it because the native interface doesn't have it) and we should be able to get it by casting `InputStream` returned by this method to `MultipartInputStream` and then getting length.